conan/tools/cmake/toolchain/blocks.py

```diff
@@ -667,6 +667,30 @@ class TryCompileBlock(Block):
         """)
 
 
+class CompilersBlock(Block):
+    template = textwrap.dedent(r"""
+        {% for lang, compiler_path in compilers.items() %}
+        set(CMAKE_{{ lang }}_COMPILER "{{ compiler_path|replace('\\', '/') }}")
+        {% endfor %}
+    """)
+
+    def context(self):
+        # Reading configuration from "tools.build:compiler_executables" -> {"C": "/usr/bin/gcc"}
+        compilers_by_conf = self._conanfile.conf.get("tools.build:compiler_executables", default={},
+                                                     check_type=dict)
+        # Map the possible languages
+        compilers = {}
+        compiler_launchers = {}
+        # Allowed <LANG> variables (and <LANG>_LAUNCHER)
+        compilers_mapping = {"cc": "C", "cuda": "CUDA", "cxx": "CXX",
```

Comment:
Maybe I would use ``c`` instead of ``cc`` and ``cpp`` instead of ``cxx``. I think it's also worth adding something like:
`fortran` -> `Fortran`
`asm` -> `ASM`